MACServerExploit.py

```diff
@@ -60,8 +60,11 @@ def __init__(self, mac):
     
     def __recv_manager__(self):
         while self.work:
-            data, _ = self.sock.recvfrom(1024*64)
-            self.__parse_packet__(data)
+            try:
+                data, _ = self.sock.recvfrom(1024*64)
```

Comment:
This is where the `__recv_manager__` hangs and waits to receive something. In the case of a connection timeout, I don't know how to abort this.
Added a try-catch around this in any case, so if it is aborted with either timeout or shutdown it doesn't explode.